be/src/vec/sink/vmysql_result_writer.cpp

```diff
@@ -186,6 +188,58 @@ Status VMysqlResultWriter::_add_one_column(const ColumnPtr& column_ptr,
             _buffer.close_dynamic_mode();
             result->result_batch.rows[i].append(_buffer.buf(), _buffer.length());
         }
+    } else if constexpr (type == TYPE_MAP) {
+    	        auto& column_map = assert_cast<const ColumnMap&>(*column);
+        auto& offsets = column_map.get_offsets();
+        auto& column_key_array = assert_cast<const ColumnArray&>(column_map.get_keys());
+        auto& column_val_array = assert_cast<const ColumnArray&>(column_map.get_values());
+        auto& map_type = assert_cast<const DataTypeMap&>(*nested_type_ptr);
+        auto& key_nested_type_ptr = map_type.get_key_type();
+        auto& val_nested_type_ptr = map_type.get_value_type();
+        for (ssize_t i = 0; i < row_size; ++i) {
+            if (0 != buf_ret) {
+                return Status::InternalError("pack mysql buffer failed.");
+            }
+            _buffer.reset();
+
+            _buffer.open_dynamic_mode();
+            buf_ret = _buffer.push_string("{", 1);
```

Comment:
can DataTypeMap::to_string() be reused here to reduce redundant code ?

Reply:
Yes . I will